### PR TITLE
Allow command line args to modify options

### PR DESCRIPTION
**Pull request by samshull**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -44,6 +44,14 @@ module.exports = function (gulp, options) {
     mocha: {}
   });
 
+  // allows command line arguments to modify the options object
+  // ex: gulp test --mocha.grep @production-only --mocha.invert
+  // -- options.mocha == { grep: '@production-only', invert: true }
+  _.each(argv, (value, name) => {
+    if (name === '_' || name === '$0') return;
+    options[name] = _.merge(options[name] || {}, value);
```
Comment (DullReferenceException) [inline lib/index.js:52]: This is only going one level deep; it'd be a more complete feature if you could override options at any level. Also, it'd be useful to have this command line override ability documented in `README.md`.

**Review by DullReferenceException — CHANGES_REQUESTED** [review]

File: lib/index.js (hunk 1 above)
Comment (DullReferenceException) [reply lib/index.js:52]: Never mind, looks like you're parsing the command line with yargs, so that's where the magic happens.

**Comment by DullReferenceException** [comment]
:+1:

**Comment by Swaagie** [comment]
+1

**Pull request merged**